sklearn/datasets/tests/test_common.py

```diff
@@ -0,0 +1,10 @@
+"""Test loaders for common functionality.
+"""
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_true
+
+
+def check_return_X_y(bunch, X_y_tuple):
```

Comment:
I had imagined this would take a partial function and pass in return_X_y=True. You still have left a lot of duplicated idiom in the test functions

Reply:
By 

> a lot of duplicated idiom

do you mean the setting up of the bunches and fetch of the X_y_tuples to compare?

Were you thinking of something like this?

**Each of the dataset test files sets up the appropriate fetch partial**

* test_20news.py, test_lfw.py, etc.

```python
from functools import partial
# test return_X_y option
fetch_func = partial(datasets.fetch_20newsgroups_vectorized, subset='test')
check_return_X_y(bunch, fetch_func)
```

**During the check_return_X_y call, the partial is called, passing return_X_y=True** 

* test_common.py

```python
def check_return_X_y(bunch, fetch_func_partial):
    X_y_tuple = fetch_func_partial(return_X_y=True)
    assert(isinstance(X_y_tuple, tuple))
    assert(X_y_tuple[0].shape == bunch.data.shape)
    assert(X_y_tuple[1].shape == bunch.target.shape)
```

Wouldn't this run into problems if ever additional arguments are added to those fetch functions after what's currently the last parameter, the return_X_y one?